modules/infra/control/vlan.c

```diff
@@ -129,6 +129,8 @@ static int iface_vlan_reconfig(
 	if (set_attrs & GR_IFACE_SET_VRF)
 		iface->vrf_id = vrf_id;
 
+	iface_event_notify(IFACE_EVENT_POST_RECONFIG, iface);
```

Comment:
could we get from this API the iface state before and the newly iface state ?

Reply:
What do you mean by "from the API"?

Reply:
We need a third argument with the iface before. I know that it seems painful but it'll help the management planes and control planes to diff it.

Reply:
What about having a IFACE_EVENT_PRE_RECONFIG in addition to POST ?
This wouldn't require any modification in the API, and we may be able to get this diff.

Reply:
OK, then with a transaction ID (whatever uuid or counter) that shall be the same for the IFACE_EVENT_PRE_RECONFIG and then IFACE_EVENT_POST_RECONFIG so the management plan can correlate both.

Reply:
The PRE_RECONFIG event and the transition id may not be required at all.
One of the assumption is that the mgmt app will be on the same node (grout uses a unix socket for its configuration).
When the management app connects, it reads the *current* state of the interfaces, so it only needs an event and the associated iface.

So, it can make a diff between the known state and the new state (hence the POST_RECONFIF notification).
WDYT ?

I have an ugly code snippet with FRR, for a dplane plugin: (Far from being ready, I just use it to validate that the notification system is working and useful. )
```
	if(gr_api_client_recv_notification(grout_ctx.notifs, &n) == 0) {
		switch (n->type) {
		case IFACE_EVENT_POST_ADD:
		case IFACE_EVENT_STATUS_UP:
		case IFACE_EVENT_STATUS_DOWN:
		case IFACE_EVENT_POST_RECONFIG:
			if (n->payload_len == sizeof(*p)) {
				p = (struct gr_infra_iface_get_resp *)&n[1];
				zlog_debug("Iface %s: %s", evt_to_str(n->type) , p->iface.name);
				iface = if_get_by_name(p->iface.name, p->iface.vrf_id, NULL);
				if(iface == NULL)
					break;
				sync_iface_status(iface, &p->iface);
			};
			break;
		case IFACE_EVENT_PRE_REMOVE:
			if (n->payload_len == sizeof(*p)) {
				p = (struct gr_infra_iface_get_resp *)&n[1];
				zlog_debug("Iface %s: %s", evt_to_str(n->type) , p->iface.name);
				iface = if_get_by_name(p->iface.name, p->iface.vrf_id, NULL);
				if(iface == NULL)
					break;
				if_delete(&iface);
			};
			break;
		default:
			zlog_debug("Unknown notification 0x%x received", n->type);
			break;
		}
```